### ISSUE TITLE
boost compute partial_sort

### ISSUE THREAD

**Issue by WilliamTambellini** [boostorg/compute#757]
Would it make sens to implement a boost::compute::partial_sort ?
Pure Cpp std::partial_sort : 
http://en.cppreference.com/w/cpp/algorithm/partial_sort
Looping by calling several nth_element() could be a workaround I guess.
Kind

**Comment by kylelutz** [comment]
Yeah, we should be able to provide ` partial_sort()` implementation. Are you interested in submitting a pull request for this?

**Comment by WilliamTambellini** [comment]
Hi @kylelutz 
I dont plan to implement it by myself but could help for testing/benchmarking. I m mainly interested by Intel CPU backends. 
Would the radix select from viennaCL be good enough : 
https://github.com/vpa1977/smalltests/blob/master/radix_select.hpp
?
kind
